Assets/Scripts/Singletons/SoundManager.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using UnityEngine;
+
+namespace ProjectWendigo
+{
+    public class SoundManager : MonoBehaviour
+    {
+        [SerializeField] private Sound[] Sounds;
+
+        /// <summary>
+        /// Get an AudioSource for the corresponding sound clip.
+        /// </summary>
+        /// <param name="name">Sound clip name from the SoundManager sounds array</param>
+        /// <returns>New AudioSource with the corresponding sound clip if found</returns>
+        public AudioSource GetAudio(string name)
+        {
+            Sound sound = this.GetSound(name);
+            if (sound == null)
+                return null;
+            return this.SpawnAudio(this.gameObject, sound);
+        }
+
+        /// <summary>
+        /// Get an AudioSource positioned at a given position in world space.
+        /// </summary>
+        /// <param name="name">Sound clip name from the SoundManager sounds array</param>
+        /// <param name="position">Position in world space where the sound should be emitted from</param>
+        /// <param name="spatialBlend">How much the AudioSource is affected by 3D spatialisation calculations</param>
+        /// <returns>New AudioSource with the corresponding sound clip if found</returns>
+        public AudioSource GetAudioAt(string name, Vector3 position, float spatialBlend = 1f)
+        {
+            Sound sound = this.GetSound(name);
+            if (sound == null)
+                return null;
+            var gameObject = new GameObject(name);
+            gameObject.transform.position = position;
+            gameObject.transform.parent = this.transform;
```

Comment:
Why?

Reply:
**Why I attach the generated game object to the sound manager?**
To avoid polluting the scene hierarchy, I prefer to attach all generated game objects to the sound manager. It's arbitrary but doesn't change the intended behavior.

**Why I do it like that?**
That's just Unity's way of doing it, you can only access a game object's parent/children through its transform property...
https://answers.unity.com/questions/989146/how-to-attach-an-object-onto-another-object.html

Reply:
Wow Unity makes so much sense, that's awesome!